src/engine/DisjunctionConstraint.cpp

```diff
@@ -182,6 +183,85 @@ PiecewiseLinearCaseSplit DisjunctionConstraint::getValidCaseSplit() const
     return getImpliedCaseSplit();
 }
 
+void DisjunctionConstraint::transformIfNeeded( InputQuery &inputQuery )
+{
+    Vector<PiecewiseLinearCaseSplit> newDisjuncts;
+    for ( const auto &disjunct : _disjuncts )
+    {
+        PiecewiseLinearCaseSplit newDisjunct;
+
+        // Store the bounds in the old disjunct
+        for ( const auto &bound : disjunct.getBoundTightenings() )
+            newDisjunct.storeBoundTightening( bound );
+
+        List<Equation> equationsToProcess;
+        for ( const auto &equation : disjunct.getEquations() )
+        {
+            if ( equation._type == Equation::EQ )
+            {
+                Equation equation1 = equation;
+                equation1._type = Equation::GE;
+                Equation equation2 = equation;
+                equation2._type = Equation::LE;
+                equationsToProcess.append( equation1 );
+                equationsToProcess.append( equation2 );
+            }
+            else
+            {
+                equationsToProcess.append( equation );
+            }
+        }
+
+        /*
+          Given a constraint AX >= b in the disjunct, we introduce an auxiliary
+          variable aux, add new linear constraints AX + aux = b and change the
+          constraint in the disjunct to aux <= 0
+
+          The LE case is symmetric.
+        */
+        for ( const auto &equation : equationsToProcess )
+        {
+            unsigned aux = inputQuery.getNumberOfVariables();
```

Comment:
Lines 224-225 are probably a common pattern that could be a new function in InputQuery. Although this PR might not be the place to do it.

Reply:
Yeah, these two lines are called in different places. Maybe worth doing a pass in a separate PR.